mycroft/skills/__main__.py

```diff
@@ -12,214 +12,314 @@
+
+    def _start_intent_services(self):
+        """Instantiate the Adapt and Padatious intent managers.
+
+        The intent managers convert utterances to intents, which used to invoke
+        the skill system.  Skill handlers define intents that, when matched,
+        will execute the code within the skill handler.
+
+        When no match is found for the intent, an intent_failu re message is
+        emitted, which invokes the fallback mechanism.
+        """
+        LOG.info('Starting Adapt and Padatious intent services...')
+        service = IntentService(self.bus)  # Adapt
+        try:
+            PadatiousService(self.bus, service)
+        except Exception as e:
+            LOG.error('Failed to create Padatious intent handlers')
+            LOG.exception(e)
+
+    def _start_skill_manager(self):
+        """Create a thread that monitors loaded skills looking for updates."""
+        LOG.info('Starting the skill manager...')
+        # TODO: if this needs an internet connection, don't start it so soon
+        try:
+            # self.skill_manager = SkillManager(self.bus, self.config)
+            self.skill_manager = SkillManager(self.bus)
+        except MsmException:
+            # skill manager couldn't be created, wait for network connection
+            # and retry
+            LOG.error(
+                'Msm is uninitialized and requires network connection to fetch'
+                'skill information\n Waiting for network connection...'
+            )
+            while not connected():
+                time.sleep(30)
+            # self.skill_manager = SkillManager(self.bus, self.config)
+            self.skill_manager = SkillManager(self.bus)
+
+        self.skill_manager.daemon = True
+
+        # TODO: do this before connection check but handle a failure before?!?
+        # Wait until priority skills have been loaded before checking
+        # network connection
+        self.skill_manager.load_priority()
+        # If device has been offline for more than 2 weeks or
+        # skill state is in a limbo do a quick update
+        update_limit = datetime.now() - timedelta(days=14)
+        if (not self.skill_manager.last_download or
+                self.skill_manager.last_download < update_limit):
+            self.skill_manager.download_skills(quick=True)
+        else:  # Just post the skills manifest
+            self.skill_manager.post_manifest(self.skill_manager.msm)
+        self.skill_manager.start()
+
+
+class DevicePrimer(object):
+    def __init__(self, message_bus_client, config):
+        self.bus = message_bus_client
+        self.platform = config['enclosure'].get("platform", "unknown")
+        self.enclosure = None
+        self.is_paired = False
+        self.backend_down = False
+
+    def prepare_device(self):
+        """Internet dependent updates of various aspects of the device."""
+        self.enclosure = EnclosureAPI(self.bus)
+        self._get_pairing_status()
```

Comment:
On reordering the code, I didn't want to do anything with the backend down exception until after the series of events that need the backend start.  For example, if the system is going to restart due to time skew, it really does not good to tell the user it can't connect with the backend yet.

Reply:
I mean since the other methods basically require that `_get_pairing_status()` has been called so that the members created here exists/is set and there's no indication of the required order it would be good if it's done as needed.

Reply:
There is one other thing that concerns me about making this a property.  The method actually sets two instance attributes.  I don't like the idea of a property that has a side-effect.  Properties should do one thing only, and that is set the value of that property.

I understand your point about other methods requiring it but there is only one public method and it forces the order of the method calls so I think we are OK there.

Reply:
Fair point about the side effect, that's not pretty.